src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -32,56 +32,82 @@ namespace Microsoft.Diagnostics.Tracing.Analysis
     /// </summary>
     public static class TraceLoadedDotNetRuntimeExtensions
     {
+        internal static readonly string LoadedDotNetRuntimesKey = "Computers/LoadedDotNetRuntimes";
+
         public static void NeedLoadedDotNetRuntimes(this TraceEventDispatcher source)
         {
+            if (source.userData.ContainsKey(LoadedDotNetRuntimesKey))
+            {
+                // The callbacks are already registered for this source
+                return;
+            }
+
             // ensure there are base processes
```

Comment:
This change is not resilient to calling source.Process twice.  The data in DotNetRuntime is accumulative and if not cleared, will be wrong.